### PR TITLE
Add usage analytics

### LINKED ISSUE #236: Add usage analytics
We currently don't have good insight on how (or if) are users are using the app. We should be able to track what users are using it in production, how frequently they use it, etc.

While we're in here making this change, we should go ahead and add version numbering to the footer or somewhere so we can easily check what code is deployed

### PR DESCRIPTION
**Pull request by mddilley**
Closes #236

### CODE REVIEW

**Comment by mddilley** [comment]
@mateoclarke - this PR is ready to go except for the Segment.io code in `index.html` needs to be updated after creating a DTS account (it is currently tied to my email). The free tier allows tracking up to 1000 users. I added the app version to display in the `UserInfo`(logout) component, and the rest of the changes identify the user and the pages navigated by the user. I can walk through the stats when we get a chance.

### App version
Screen Shot 2019-12-10 at 5 37 07 PM

File: src/components/UserInfo.js [hunk 1]
```diff
@@ -52,13 +54,18 @@ class UserInfo extends Component {
...
+              <DropdownItem>
+                <Button
+                  color="danger"
+                  onClick={this.props.revokeKnackUserToken}
+                  className="btn-lg mt-1"
+                >
+                  Log out
+                </Button>
+                <span className="align-bottom font-weight-light ml-2">
+                  v{pckg.version}
```
Comment (mateoclarke) [inline src/components/UserInfo.js]: 👍

File: public/index.html [hunk 2]
```diff
@@ -11,6 +11,17 @@
...
+
+  <script>
+    !function () {
+      var analytics = window.analytics = window.analytics || []; if (!analytics.initialize) if (analytics.invoked) window.console && console.error && console.error("Segment snippet included twice."); else {
+        analytics.invoked = !0; analytics.methods = ["trackSubmit", "trackClick", "trackLink", "trackForm", "pageview", "identify", "reset", "group", "track", "ready", "alias", "debug", "page", "once", "off", "on"]; analytics.factory = function (t) { return function () { var e = Array.prototype.slice.call(arguments); e.unshift(t); analytics.push(e); return analytics } }; for (var t = 0; t < analytics.methods.length; t++) { var e = analytics.methods[t]; analytics[e] = analytics.factory(e) } analytics.load = function (t, e) { var n = document.createElement("script"); n.type = "text/javascript"; n.async = !0; n.src = "https://cdn.segment.com/analytics.js/v1/" + t + "/analytics.min.js"; var a = document.getElementsByTagName("script")[0]; a.parentNode.insertBefore(n, a); analytics._loadOptions = e }; analytics.SNIPPET_VERSION = "4.1.0";
+        analytics.load("mNZ52QK4RGH7GyKot2UGGyj5lCrS1eNL");
```
Comment (mateoclarke) [inline public/index.html]: i'm guessing this is the part of the code that needs to be updated with a new ID once we set up a Segment account for the team.

Comment (mateoclarke) [reply public/index.html]: Wondering if we need to do any environment checking, we probably don't want to be logging user data from staging.

Comment (mddilley) [reply public/index.html]: That is a good consideration. I'll do some testing. I think setting the url in the Segment should filter out the staging visits, but it is possible that the app ID in the JS is enough generate data.

Comment (mddilley) [reply public/index.html]: After some testing, the analytics are tracked even in development. I added a condition to set the app ID to empty string if host is `localhost` and that works. I tried not loading analytics at all, but then the calls to analytics.js in the React components break the app.

**Pull request merged**